File: matiasmasso/Mat.NET/LegacyHelper/AsinHelper.vb
Public Class AsinHelper
    Shared Function GenerateRandomASIN() As String
        Dim s As String = ""
        Dim r As New Random()
        Dim i As Integer
        For i = 1 To 10
            Dim c As Char = ChrW(r.Next(65, 91))
            s = s & c
        Next
        Return s
    End Function
End Class

File: matiasmasso/DTO/Helpers/ASINHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DTO.Helpers
{
    public class ASINHelper
    {
        private static Random random = new Random();

        public static string GenerateRandomASIN()
        {
            const string chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
            return new string(Enumerable.Repeat(chars, 10)
                .Select(s => s[random.Next(s.Length)]).ToArray());
        }
    }
}
